c++/src/kj/common.h

```diff
@@ -1928,6 +1928,38 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
     return true;
   }
 
+  inline bool operator<(const ArrayPtr& other) const {
+    size_t comparisonSize = kj::min(size_, other.size_);
+    if constexpr (isIntegral<RemoveConst<T>>()) {
+      int ret = memcmp(ptr, other.ptr, comparisonSize * sizeof(T));
+      if(ret != 0){
+        return ret < 0;
+      }
+    }
+    else {
+      for(size_t i = 0; i < comparisonSize; i++) {
+        bool ret = ptr[i] == other.ptr[i];
+        if(!ret) {
+          return ptr[i] < other.ptr[i];
+        }
+      }
+    }
+    return size_ <  other.size_;
+  }
+
+  inline bool operator<=(decltype(nullptr)) const { return size_ == 0; }
```

Comment:
I wonder if all these additional `nullptr` declarations are really needed? nullptr is convertible to ArrayPtr, no?